### ISSUE TITLE
JGI depth file error

### ISSUE THREAD

**Issue by rgtaketani** [RasmussenLab/vamb#96]
Dear all,

I am trying to run vamb and I was having problems inputing bam files [previouly](https://github.com/RasmussenLab/vamb/issues/51). Thus, I decided to use the JGI route, however vam is still giving me errors:
```
Traceback (most recent call last):
  File "/home/data/mmeg_ngs/rodrigo_taketani/condaenvs/vamb/bin/vamb", line 11, in <module>
    sys.exit(main())
  File "/home/data/mmeg_ngs/rodrigo_taketani/condaenvs/vamb/lib/python3.6/site-packages/vamb/__main__.py", line 476, in main
    logfile=logfile)
  File "/home/data/mmeg_ngs/rodrigo_taketani/condaenvs/vamb/lib/python3.6/site-packages/vamb/__main__.py", line 219, in run
    len(tnfs), minalignscore, minid, subprocesses, logfile)
  File "/home/data/mmeg_ngs/rodrigo_taketani/condaenvs/vamb/lib/python3.6/site-packages/vamb/__main__.py", line 103, in calc_rpkm
    rpkms = vamb.vambtools.load_jgi(file)
  File "/home/data/mmeg_ngs/rodrigo_taketani/condaenvs/vamb/lib/python3.6/site-packages/vamb/vambtools.py", line 407, in load_jgi
    raise ValueError('Input file format error: 5th column should contain variances.')
ValueError: Input file format error: 5th column should contain variances.
```
I checked the depth file and it seems like the variance is there in the 5th column:
```
:head /home/data/mmeg_ngs_seqs_database/MAG/depth_vamb.txt
contigName      contigLen       totalAvgDepth   map_to_vamb_sorted.bam  map_to_vamb_sorted.bam-var
k141_0  419     86.7509 86.7509 27459.4
k141_10813757   393     2.16049 2.16049 0.829506
k141_11972372   348     3.69192 3.69192 1.10765
k141_386206     472     2.31056 2.31056 0.987366
k141_5793090    474     2.90741 2.90741 0.480565
k141_11199962   489     5.57227 5.57227 2.34609
k141_8496527    312     1.60494 1.60494 0.240473
k141_2317236    495     4.66087 4.66087 3.61431
k141_11586167   427     1.89892 1.89892 0.402789
```

**Comment by jakobnissen** [comment]
What version are you running? I just checked your sample file and it worked on version 3.0.2

**Comment by rgtaketani** [comment]
@jakobnissen I am running version 2.1.0. I am going update it to the most recent version and try it again. I don't understand why conda installed an old version.

**Comment by simonrasmu** [comment]
Conda sometimes installs the old version of VAMB because of your python installation. It works with python=3.7 for me

**Comment by hhf95** [comment]
Hi Jakob

I had a same error "ValueError: Length of TNFs and length of RPKM does not match. Verify the inputs" when I run vamb 3.0.2

my commands is: (--minfasta added or not give me the same error)

       vamb --outdir output--fasta contigs.fa --jgi np.cov  (--minfasta 200000)

head of jgi file is

      contigName      contigLen       totalAvgDepth   np.cov.bam
      contig_1        217016  13.7727 13.7727
      contig_1000     3099    172.475 172.475
      contig_10002    1233    12.759  12.759
      contig_10003    11265   13.2576 13.2576
      contig_10005    447751  153.083 153.083
      contig_10006    30167   3.85558 3.85558
      contig_10007    4322    4.95206 4.95206
      contig_10008    7869    5.2008  5.2008
      contig_1001     3533    91.6834 91.6834


Here is my log txt:

      Starting Vamb version 3.0.2
              Date and time is 2021-12-14 16:23:13.206712
      
      Loading TNF
              Minimum sequence length: 100
              Loading data from FASTA file contigs.fa
      
              Kept 2703388420 bases in 62451 sequences
              Processed TNF in 24.21 seconds
      
      Loading RPKM
              Loading RPKM from JGI file ./np.cov


It seems TNF calculating filtering contigs small than 100? or what the reason cause the error?

**Comment by jakobnissen** [comment]
@hhf95 Indeed, in version 3.0.2, JGI parsing did not filter contig lengths. This has been fixed in 3.0.3.
At some point soon I'll figure out how to update pip and Conda to version 3.0.3, but Python package mangement is a total mess, so let's see.

**Comment by jakobnissen** [comment]
Closing this issue as solved, although I'll need to update the Conda package soon.
